domains/grid/components/AddWidgetText.vue

```diff
@@ -0,0 +1,230 @@
+})
+const inputErrors = reactive({
+  text: '',
+  titleColor: '',
+  textColor: '',
+  backgroundColor: '',
+})
+
+const canSubmit = ref(false)
+const isEdit = computed(() => !!widgetData.value)
+
+const handleSave = () => {
+  if (!canSubmit.value) {
+    return
+  }
+
+  const properties = toRaw(inputValues)
+
+  if (isEdit.value) {
+    const updatedWidget: GridWidget = {
+      ...(widgetData.value as GridWidget),
+      properties,
+    }
+    updateGridLayoutItem(updatedWidget)
+  } else {
+    const newWidget: GridWidgetWithoutCords = createWidgetObject({
+      type: GRID_WIDGET_TYPE.TEXT,
+      properties,
+    })
+
+    addGridLayoutItem(newWidget)
+  }
+
+  handleCancel()
+}
+
+const handleCancel = () => {
+  clearWidgetData()
+  closeModal()
+}
+
+const handleTitleChange = async (customEvent: CustomEvent) => {
+  const event = customEvent.detail.event
+  const input = event.target as HTMLInputElement
+  inputValues.title = input.value
+}
+
+const handleTextChange = async (customEvent: CustomEvent) => {
+  const event = customEvent.detail.event
+  const input = event.target as HTMLInputElement
+  inputErrors.text = ''
+
+  if (!input.value) {
+    inputErrors.text = formatMessage('errors_required')
+    return
+  }
+
+  inputValues.text = input.value
+}
+
+const handleTitleColorChange = async (customEvent: CustomEvent) => {
+  const event = customEvent.detail.event
+  const color = event.target as HTMLInputElement
+  const tColor = tinycolor(color.value)
+  inputErrors.titleColor = ''
+
+  if (!tColor.isValid()) {
+    inputErrors.titleColor = formatMessage('errors_invalid_hex_color')
+    return
+  }
+
+  inputValues.titleColor = color.value
+}
+
+const handleTextColorChange = async (customEvent: CustomEvent) => {
+  const event = customEvent.detail.event
+  const color = event.target as HTMLInputElement
+  const tColor = tinycolor(color.value)
+  inputErrors.textColor = ''
+
+  if (!tColor.isValid()) {
+    inputErrors.textColor = formatMessage('errors_invalid_hex_color')
+    return
+  }
+
+  inputValues.textColor = color.value
+}
+
+const handleBackgroundColorChange = async (customEvent: CustomEvent) => {
+  const event = customEvent.detail.event
+  const color = event.target as HTMLInputElement
+  const tColor = tinycolor(color.value)
+  inputErrors.backgroundColor = ''
+
+  if (!tColor.isValid()) {
+    inputErrors.backgroundColor = formatMessage('errors_invalid_hex_color')
+    return
+  }
+
+  inputValues.backgroundColor = color.value
+}
+
+watchEffect(() => {
+  canSubmit.value =
+    !!inputValues.text &&
+    !inputErrors.text &&
+    !inputErrors.titleColor &&
+    !inputErrors.textColor &&
+    !inputErrors.backgroundColor
+})
+
+onMounted(() => {
+  setTimeout(() => {
+    const input = document?.querySelector(
+      'lukso-input'
+    ) as unknown as HTMLElement
+
+    input?.shadowRoot?.querySelector('input')?.focus()
+  }, INPUT_FOCUS_DELAY)
+
+  Object.assign(inputValues, widgetData.value?.properties)
```

Comment:
Wouldnt here the `cloneObject` make sense?

Reply:
It's different case. Here we are not making copy of an object but we want to set widget properties on `inputValues` reactive property while we are in edit mode.